youtube_dl/extractor/rai.py

```diff
@@ -500,3 +502,108 @@ def _real_extract(self, url):
         info.update(relinker_info)
 
         return info
+
+
+class HTMLListAttrsParser(compat_HTMLParser):
+    def __init__(self):
+        compat_HTMLParser.__init__(self)
+        self.items = []
+        self._level = 0
+
+    def handle_starttag(self, tag, attrs):
+        if tag == 'li' and self._level == 0:
+            self.items.append(dict(attrs))
+        self._level += 1
+
+    def handle_endtag(self, tag):
+        self._level -= 1
+
+
+class RaiPlayRadioBaseIE(InfoExtractor):
+    _BASE = 'https://www.raiplayradio.it'
+
+    def parse_list(self, webpage):
+        parser = HTMLListAttrsParser()
+        parser.feed(webpage)
+        parser.close()
+        return parser.items
+
+    def get_playlist_iter(self, url):
+        webpage = self._download_webpage(url, url)
+        for attrs in self.parse_list(webpage):
+            title = attrs['data-title'].strip()
+            webpage = urljoin(url, attrs['data-mediapolis'])
+            audio_url = compat_str(self._request_webpage(
+                webpage, title).geturl())
+            yield {
+                'url': audio_url,
+                'id': attrs['data-uniquename'].lstrip('ContentItem-'),
+                'title': title,
+                'ext': determine_ext(audio_url),
+                'thumbnail': urljoin(url, attrs['data-image']),
+                'language': 'it'}
+
+    def get_playlist(self, *args, **kwargs):
+        return list(self.get_playlist_iter(*args, **kwargs))
+
+
+class RaiPlayRadioIE(RaiPlayRadioBaseIE):
+    _VALID_URL = r'%s/audio/.+?-(?P<id>%s)\.html' % (
+        RaiPlayRadioBaseIE._BASE, RaiBaseIE._UUID_RE)
+    _TEST = {
+        'url': 'https://www.raiplayradio.it/audio/2019/07/'
+               'RADIO3---LEZIONI-DI-MUSICA-'
+               '36b099ff-4123-4443-9bf9-38e43ef5e025.html',
```

Comment:
Are you trolling or what?
https://github.com/ytdl-org/youtube-dl#long-lines-policy
>For example, you should **never** split long string literals like URLs

Reply:
No trolling of course. The PR had URL on the same line, but after not having feedback except "read the conventions" I double checked again trying to figure out what was wrong and I probably overlooked that.
Fixed.